src/Services/PortfolioPlanningDataService.ts

```diff
@@ -256,44 +210,39 @@ export class PortfolioPlanningDataService {
 
         allPlans.entries.push(savedPlan);
 
-        await client.updateDocument(
-            PortfolioPlanningDataService.DirectoryCollectionName,
-            allPlans
-        );
+        await client.updateDocument(PortfolioPlanningDataService.DirectoryCollectionName, allPlans);
 
         return newPlan;
     }
 
-    public async GetPortfolioPlanById(
-        portfolioPlanId: string
-    ): Promise<PortfolioPlanning> {
+    public async GetPortfolioPlanById(portfolioPlanId: string): Promise<PortfolioPlanning> {
         const client = await this.GetStorageClient();
         const planIdLowercase = portfolioPlanId.toLowerCase();
 
         return client.getDocument(PortfolioPlanningDataService.PortfolioPlansCollectionName, planIdLowercase);
     }
 
-    public async UpdatePortfolioPlan(
-        newPlan: PortfolioPlanning
-    ): Promise<PortfolioPlanning> {
+    public async UpdatePortfolioPlan(newPlan: PortfolioPlanning): Promise<PortfolioPlanning> {
         const client = await this.GetStorageClient();
-        
+
         //  TODO    sanitize other properties (e.g. unique set of work item ids, all strings lower case)
         newPlan.id = newPlan.id.toLowerCase();
 
         return client
-            .updateDocument(
-                PortfolioPlanningDataService.PortfolioPlansCollectionName,
-                newPlan
-            )
+            .updateDocument(PortfolioPlanningDataService.PortfolioPlansCollectionName, newPlan)
             .then(doc => doc);
     }
 
-    public async DeletePortfolioPlan(
-        newPlan: PortfolioPlanning
-    ): Promise<void> {
+    public async DeletePortfolioPlan(planId: string): Promise<void> {
         const client = await this.GetStorageClient();
-        const planIdToDelete = newPlan.id.toLowerCase();
+        const planIdToDelete = planId.toLowerCase();
+
+        let allPlans = await this.GetAllPortfolioPlans();
+
+        const indexToDelete = allPlans.entries.findIndex(plan => plan.id == planIdToDelete);
```

Comment:
not supported in IE... you could use `filter` to exclude the plan to be deleted, and the use the result to update the document.